tests/python/gaia-ui-tests/gaiatest/tests/functional/browser/test_browser_launch_from_background.py

```diff
@@ -0,0 +1,37 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from gaiatest import GaiaTestCase
+from marionette_driver import By, Wait
+
+from gaiatest.apps.search.app import Search
+from gaiatest.apps.search.regions.browser import Browser
+
+
+class TestLaunchViaManifest(GaiaTestCase):
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        self.connect_to_local_area_network()
+
+        self.test_url = self.marionette.absolute_url('mozilla.html')
+
+    def test_launch_manifest(self):
+        search = Search(self.marionette)
+        search.launch()
+
+        self.device.touch_home_button()
+
+        search.launch()
+
+        browser = search.go_to_url(self.test_url)
+        browser.switch_to_content()
+        Wait(self.marionette).until(lambda m: m.title == 'Mozilla')
+
+        self.device.touch_home_button()
+
+        search.launch()
```

Comment:
Ok, I will do this. Note that this will pollute browser locators to the search app.
But I think this whole distinction between Browser and Search is not really logical, they are almost the same.
I'll file a bug on removing the distinction between the two.